web3_proxy/src/frontend/admin.rs

```diff
@@ -33,6 +39,117 @@ use std::sync::Arc;
+
+    // Check if the caller is an admin (if not, return early)
+    let admin_entry: admin::Model = admin::Entity::find()
+        .filter(admin::Column::UserId.eq(caller_id))
+        .one(&db_conn)
+        .await?
+        .ok_or(Web3ProxyError::AccessDenied)?;
+
+    // println!("Got here 2, {:?}", caller_id);
+    // Get the user from params
+    let user_address: Address = params
+        .get("user_address")
+        .ok_or_else(|| {
+            Web3ProxyError::BadRequest("Unable to find user_address key in request".to_string())
+        })?
+        .parse::<Address>()
+        .map_err(|_| {
+            Web3ProxyError::BadRequest("Unable to parse user_address as an Address".to_string())
+        })?;
+    let user_address_bytes: Vec<u8> = user_address.clone().to_fixed_bytes().into();
+    // Get the amount from params
+    let amount: Decimal = Decimal::from_str(params.get("amount").ok_or_else(|| {
+        Web3ProxyError::BadRequest("Unable to get the amount key from the request".to_string())
+    })?)
+    .or_else(|err| {
+        Err(Web3ProxyError::BadRequest(format!(
+            "Unable to parse amount from the request {:?}",
+            err
+        )))
+    })?;
+
+    // TODO: Gotta check if this reference is ok
+    let user_entry: user::Model = user::Entity::find()
+        .filter(user::Column::Address.eq(user_address_bytes.clone()))
+        .one(&db_conn)
+        .await?
+        .ok_or(Web3ProxyError::BadRequest(
+            "No user with this id found".to_string(),
+        ))?;
+
+    let increase_balance_receipt = admin_increase_balance_receipt::ActiveModel {
+        amount: sea_orm::Set(amount),
+        admin_id: sea_orm::Set(admin_entry.id),
+        deposit_to_user_id: sea_orm::Set(user_entry.id),
+        ..Default::default()
+    };
+    increase_balance_receipt.save(&db_conn).await?;
+
+    let mut out = HashMap::new();
+    out.insert(
+        "user",
+        serde_json::Value::String(format!("{:?}", user_address)),
+    );
+    out.insert("amount", serde_json::Value::String(amount.to_string()));
+
+    let txn = db_conn.begin().await?;
+
+    // Get the balance row
+    let balance_entry: balance::Model = balance::Entity::find()
+        .filter(balance::Column::UserId.eq(user_entry.id))
+        .one(&txn)
+        .await?
+        .context("User does not have a balance row")?;
+
+    let new_amount = balance_entry.available_balance + amount;
+    let mut balance_entry = balance_entry.into_active_model();
+    balance_entry.available_balance = sea_orm::Set(new_amount);
+
+    println!("New amount: {:?}", new_amount);
+
+    // Finally make the user premium if balance is above 10$
+    let premium_user_tier = user_tier::Entity::find()
+        .filter(user_tier::Column::Title.eq("Premium"))
+        .one(&txn)
+        .await?
+        .context("Premium tier was not found!")?;
+    if new_amount >= Decimal::new(10, 0) && user_entry.user_tier_id != premium_user_tier.id {
```

Comment:
I think I commented about this somewhere else. Let's not actually change the user tier in the db. the db should always show the premium tier. if they are out of funds, the rust code should change to using the downgrade_tier_id isntead. 

if you change the database, then once we start adding custom tiers for daos. when they run out of funds, putting them back to the correct tier will be a bigger challenge.

Reply:
I dont think this segment changes anything with the user-tier, no?

Reply:
ok i removed this logic completely, right now there might be a "design void" with this though, admins will have to upgrade users manually to premium